website/documentation/getting-started/common-pitfalls/_index.md

```diff
@@ -0,0 +1,204 @@
+
+{{% alert color="info"  title="Recommendation" %}}
+Control the version lifecycle separately for any cluster that hosts important workload.
+{{% /alert %}}
+
+## Node Draining 
+
+### Node Draining and Pod Disruption Budget
+
+![](./images/node-draining.gif)
+
+Typically, nodes are drained when:
+- There is a update of the OS / Kubernetes minor version
+- An Operator cordons & drains a node
+- The cluster-autoscaler wants to scale down
+
+Without a PodDistruptionBudget, pods will be terminated as fast as possible. If an application has 2 out of 2 replicas running on the drained node, this will probably cause availability issues.
+
+### Node Draining with PDB
+
+![](./images/node-draining-pdb.gif)
+
+PodDisruptionBudgets can help to manage a graceful node drain. However, if no disruptions are allowed there, the node drain will be blocked until it reaches a timeout. Only then will the nodes be terminated but without respecting PDB thresholds.
+
+{{% alert color="info"  title="Recommendation" %}}
+Configure PDBs and allow disruptions.
+{{% /alert %}}
+
+## Pod Resource Requests and Limits
+
+### Resource Consumption
+
+Pods consume resources and, of course, there are only so many resources available on a single node. Setting requests will make the scheduling much better, as the scheduler has more information available. 
+
+Specifying limits can help, but can also limit an application in unintended ways. A recommendation to start with:
+- Do not set CPU limits (CPU is compressible and throttling is really hard to detect)
+- Set memory limits and monitor OOM kills / restarts of workload (typically detectable by container status exit code 137 and corresponding events). This will decrease the likelihood of OOM situations on the node itself. However, for critical workloads it might be better to have uncapped growth and rather risk a node going OOM.
+
+Next, consider if assigning the workload to quality of service class `guaranteed` is needed. Again - this can help or be counterproductive. It is important to be aware of its implications. For more information, see [Pod Quality of Service Classes](https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/).
+
+Tune `shoot.spec.Kubernetes.kubeReserved` to protect the node (kubelet) in case of a workload pod consuming too much resources. It is very helpful to ensure a high level of stability.
+
+If the usage profile changes over time, the VPA can help a lot to adapt the resource requests / limits automatically.
+
+## Webhooks
+
+### User-Deployed Webhooks in Kubernetes
+
+![](./images/user-webhook.gif)
+
+By default, any request to the API server will go through a chain of checks. Let's take the example of creating a pod. When the resource is submitted to the API server, it will be checked if the user is authorized to perform this action or if the pod definition is actually valid / allowed. Additionally, there is the defaulting, like injecting the default service account if nothing else is specified.
```

Comment:
```suggestion
By default, any request to the API server will go through a chain of checks. Let's take the example of creating a pod. 
When the resource is submitted to the API server, it will be checked 
- is the user authorized to perform this action? 
- is the pod definitionactually valid?
- are the specified values allowed?
Additionally, there is the defaulting - like the injection of a the `default` service account's name, if nothing else is specified.
```